### ISSUE TITLE
.formatter.exs `import_deps` does not work for git dependencies

### ISSUE THREAD

**Issue by ejpcmac** [elixir-lang/elixir#7331]
### Environment

* Elixir & Erlang versions (elixir --version): 

    ```
    Erlang/OTP 20 [erts-9.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false]                    
    Elixir 1.6.1 (compiled with OTP 20)
    ```

* Operating system: Linux (Fedora 26)

### Current behavior

When I try to run `mix format` with an `import_deps` containing a Git dependency, it fails:

```
** (Mix) Unknown dependency :mock_uart given to :import_deps in the formatter configuration. The dependency is not listed in your mix.exs file
```

In my `mix.exs`, I have:

```elixir
{:mock_uart, git: "http://<some_internal_address>/mock_uart.git", tag: "v0.1.0", only: :test}
```

### Expected behavior

`mix format` should work as with Hex dependencies.

**Comment by josevalim** [comment]
Just too double check. Does it also happen after you run `mix deps.get`?

**Comment by ejpcmac** [comment]
> Just too double check. Does it also happen after you run `mix deps.get`?

Yes, I’ve even run `mix deps.clean --all`, then `mix deps.get` to be sure.

**Comment by wojtekmach** [comment]
I wasn't able to reproduce this on Elixir master with `{:ecto, github: "elixir-ecto/ecto"}`.

I think the problem is that your dependency has `only: :test` which means it's not available in the default `:dev` environment which is used when doing `mix format`. I think if you change the dep to be `only: [:dev, :test]` it should work.

**Comment by josevalim** [comment]
Good catch @wojtekmach. I will include the environment in the error message.

**Comment by axelson** [comment]
@josevalim/team I just ran into this error and it took me a while to realize that the reason that `mix format` wasn't able to find a dependency was because of `MIX_ENV`. After searching I came across this issue and noticed that this change was effectively reverted in https://github.com/elixir-lang/elixir/commit/30033a5d84eca3657bbb9fad2d20ed71b0535e8d

Is the team open to re-introducing this part of the error message? It would have saved me some time and I imagine it would save others time as well, especially those newer to Elixir and less familiar with the different environments.

See https://github.com/zachallaun/mneme/pull/51 for context on where I ran across this
